pkg/convert/tf_state.go

```diff
@@ -221,25 +222,27 @@ func TranslateState(info ProviderInfoSource, path string) (*plugin.ConvertStateR
 						name = fmt.Sprintf("%s-%s", name, instanceAddr.Value().AsString())
 					}
 
+					logicalName := name
+
+					// if the name starts with a digit, prepend an underscore to it
+					if len(name) > 0 && unicode.IsDigit(rune(name[0])) {
+						name = fmt.Sprintf("_%s", name)
+					}
+
 					// clean special characters from the name to make it a valid identifier
 					// replacing the special characters with underscores
-					specialRunes := []rune{
-						' ', '-', '.', '/', '\\', '$', '@', '#', '%',
-						'^', '&', '*', '(', ')', '"', '\'', '<', '>', '|',
-					}
 					name = strings.Map(func(r rune) rune {
-						for _, specialRune := range specialRunes {
-							if r == specialRune {
-								return '_'
-							}
+						if !unicode.IsLetter(r) && !unicode.IsDigit(r) {
+							return '_'
 						}
 						return r
 					}, name)
 
 					resources = append(resources, plugin.ResourceImport{
-						Type: pulumiType,
-						Name: name,
-						ID:   id,
+						Type:        pulumiType,
+						Name:        name,
+						LogicalName: logicalName,
```

Comment:
Ideally only set LogicalName if it differs from Name, keeps the import files easier to understand for simple cases.